### PR TITLE
[5.2] Create `foundation-support` and `foundation-testing` subcomponent

### PR DESCRIPTION
**Pull request by crynobone**
This would allow following components to be shared with Lumen, or any other packages that require access to relevant classes.

### CODE REVIEW

**Comment by crynobone** [comment]
There are however few problem with this, `URL::action()` is not available on Lumen but with this changes it can be access via Testing.

But at the same time, it going to reduce the amount of work trying to sync the namespace under Lumen Framework.

File: src/Illuminate/Foundation/Support/Composer.php [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+     *
+     * @var string
+     */
+    protected $workingPath;
+
+    /**
+     * Create a new Composer manager instance.
+     *
+     * @param  \Illuminate\Filesystem\Filesystem  $files
+     * @param  string  $workingPath
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Support/Composer.php]: string|null

File: src/Illuminate/Foundation/Support/composer.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            "name": "Taylor Otwell",
+            "email": "taylorotwell@gmail.com"
+        }
+    ],
+    "require": {
+        "php": ">=5.5.9",
+        "illuminate/console": "5.2.*",
+        "illuminate/contracts": "5.2.*",
+        "illuminate/support": "5.2.*",
+        "symfony/process": "3.0.*"
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Support/composer.json]: 2.8.*|3.0.*

File: src/Illuminate/Foundation/Testing/composer.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        {
+            "name": "Taylor Otwell",
+            "email": "taylorotwell@gmail.com"
+        }
+    ],
+    "require": {
+        "php": ">=5.5.9",
+        "illuminate/contracts": "5.2.*",
+        "illuminate/http": "5.2.*",
+        "symfony/css-selector": "3.0.*",
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Testing/composer.json]: `2.8.*|3.0.*`

File: src/Illuminate/Foundation/Testing/composer.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        "symfony/dom-crawler": "3.0.*"
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Testing/composer.json]: `2.8.*|3.0.*`

**Comment by GrahamCampbell** [comment]
:+1:

**Comment by GrahamCampbell** [comment]
I wonder if this stuff should actually be moved out of "Foundation"?

**Comment by GrahamCampbell** [comment]
I'm actually :-1: I think, on reflection. I don't think we should allow the testing stuff to be used outside.

**Comment by GrahamCampbell** [comment]
I'm going to create a lighter PR that doesn't require any composer packages being created.

**Comment by crynobone** [comment]
> I don't think we should allow the testing stuff to be used outside.

At the current state, lumen duplicating `Illuminate\Foundation\Testing` which make it difficult to include any improvement made in Laravel to be immediately available in Lumen.

As a result, we can also test packages to be compatible with Lumen and Laravel via Testbench packages. Currently there no way to include both framework for testing because we will get duplicate class under `Illuminate\Foundation\Testing` and `Illuminate\Foundation\Support`.

**Comment by GrahamCampbell** [comment]
> At the current state, lumen duplicating Illuminate\Foundation\Testing which make it difficult to include any improvement made in Laravel to be immediately available in Lumen.

It's possible that stuff will get deleted in Lumen 5.2 anyway.

**Comment by GrahamCampbell** [comment]
Merge conflicts. All the "Composer" class stuff needs reverting.

File: src/Illuminate/Cache/composer.json [hunk 5]
```diff
@@ -16,6 +16,7 @@
     "require": {
         "php": ">=5.5.9",
         "illuminate/contracts": "5.2.*",
+        "illuminate/foundation-support": "5.2.*",
```
Comment (GrahamCampbell) [inline src/Illuminate/Cache/composer.json]: not needed anymore

**Comment by taylorotwell** [comment]
Going to hold off on this.

**Pull request closed without merging**